src/components/ContentWrapper.jsx

```diff
@@ -16,25 +15,54 @@ import {
 } from '@ant-design/icons';
 import NotificationManager from './notification/NotificationManager';
 import initUpdateSocket from '../utils/initUpdateSocket';
+import loadPipelineStatus from '../redux/actions/experimentSettings/loadPipelineStatus';
+import PipelineRunningEmptyState from './PipelineRunningEmptyState';
+import PreloadContent from './PreloadContent';
+import Error from '../pages/_error';
 
 const { Sider, Footer } = Layout;
 const { Paragraph } = Typography;
 
 const ContentWrapper = (props) => {
+  const dispatch = useDispatch();
+
   const [collapsed, setCollapsed] = useState(true);
   const { children } = props;
+
   const router = useRouter();
-  const { experimentId } = router.query;
-  const route = router.route || '';
+  const { experimentId } = router?.query || {};
+  const route = router?.route || '';
 
-  const updateSocket = useRef(null);
+  const {
+    loading: pipelineLoading,
+    error: pipelineError,
+    status: pipelineStatus,
+  } = useSelector((state) => state.experimentSettings.pipelineStatus);
 
+  // This is used to prevent a race condition where the page would start loading immediately
+  // when the pipeline status was previously loaded. In that case, `pipelineLoading` is `false`
+  // and would be set to true only in the `loadPipelineStatus` action, the time between the
+  // two events would allow pages to load.
+  const [pipelineStatusRequested, setPipelineStatusRequested] = useState(false);
+
+  const updateSocket = useRef(null);
   useEffect(() => {
-    if (experimentId) {
-      updateSocket.current = initUpdateSocket(experimentId, (res) => { console.log(res); });
+    if (!experimentId) {
+      return;
     }
+
+    dispatch(loadPipelineStatus(experimentId));
+    updateSocket.current = initUpdateSocket(experimentId, (res) => { console.log(res); });
```

Comment:
this only prints the console log statement, do we need it?

Reply:
Also, I don't see this `updateSocket` variable used anywhere, is the plan to use it later? And how would we use it?

Reply:
> you can just call the `initUpdateSocket` function, no need for the console log

There is a reason, it is so we can test changes we make to the pipeline end-to-end until the UI updates are done.

Reply:
> Also, I don't see this `updateSocket` variable used anywhere, is the plan to use it later? And how would we use it?

That is the socket that will relay pipeline updates back to the UI or anything that is not directly associated with a work request submitted to the workers. This was created based on https://biomage.atlassian.net/browse/BIOMAGE-516 and the code was just moved right now.

Reply:
ok, all these make sense, thanks for explaining!